### PR TITLE
WIFI-792: RF Implementation

### PR DESCRIPTION
**Pull request by ralphlee3**
Dependent on https://github.com/Telecominfraproject/wlan-cloud-opensync-controller/pull/11

### CODE REVIEW

File: profile-models/src/test/java/com/telecominfraproject/wlan/profile/rf/models/RfConfigurationTests.java [hunk 1]
```diff
@@ -7,18 +7,29 @@
...
+		for (RadioType radioType : RadioType.validValues()) {
+		RfElementConfiguration rfConfig = RfElementConfiguration.createWithDefaults(radioType);
+		assertEquals("This should never be 'enabled' by default, it really interferes with our"
+				+ "cloud-based auto channel rebalancing", false, rfConfig.getAutoChannelSelection());
+		}
+	}
+	
+	@Test
+    @Ignore
+    public void autoCellSizeNeverSmallerThanMinCellSize()
```
[the remark this answers is not in the document]
Comment (ralphlee3) [reply profile-models/src/test/java/com/telecominfraproject/wlan/profile/rf/models/RfConfigurationTests.java:27]: I see, I'll remove that test

File: profile-models/src/main/java/com/telecominfraproject/wlan/profile/rf/models/RfElementConfiguration.java [hunk 2]
```diff
@@ -99,6 +100,16 @@ public static RfElementConfiguration createWithDefaults(RadioType radioType) {
     	return ret;
     }
     
+    public static RfElementConfiguration createWithDefaults(RadioType radioType, ApModel apModel) {
+		RfElementConfiguration ret = createWithDefaults(radioType);
```
Comment (christieunitt-cu) [inline profile-models/src/main/java/com/telecominfraproject/wlan/profile/rf/models/RfElementConfiguration.java:104]: RfElementConfiguration object is radio-specific. If you add radioType as a private member attribute that's set here then there won't be a need for any public member methods which have a radioType parameter. This is the same approach used in ElementRadioConfig. Specifically, it means the radioType parameter defined for getMinAutoCellSize can be removed. 

Similar static maps and getters may also be needed to replace the public DEFAULT_ constants

Comment (christieunitt-cu) [reply profile-models/src/main/java/com/telecominfraproject/wlan/profile/rf/models/RfElementConfiguration.java:104]: Are you able to add radioType as a member attribute?

Comment (ralphlee3) [reply profile-models/src/main/java/com/telecominfraproject/wlan/profile/rf/models/RfElementConfiguration.java:104]: Yes I'll add that. 
I've also adjust the logic in getMinAutoCellSize to reflect that as well.

**Pull request merged**